gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java

```diff
@@ -329,6 +355,44 @@ public void deleteMember(String api, String username) {
             throw new TechnicalManagementException("An error occurs while trying to delete member " + username + " for API " + api, ex);
         }
     }
+    
+    @Override
+    public boolean isAPISynchronized(ApiEntity api) {
+    	Set<EventEntity> events = eventService.findByType(Arrays.asList(EventType.PUBLISH_API, EventType.UNPUBLISH_API));
+    	List<EventEntity> eventsSorted = events.stream().sorted((e1, e2) -> e1.getCreatedAt().compareTo(e2.getCreatedAt())).collect(Collectors.toList());
+    	Collections.reverse(eventsSorted);
+    	try {
+    		for (EventEntity event : eventsSorted) {
+    			JsonNode node = objectMapper.readTree(event.getPayload());
+    			Api payloadEntity = objectMapper.convertValue(node, Api.class);
+    			if (api.getId().equals(payloadEntity.getId())) {
+    				return api.getUpdatedAt().compareTo(payloadEntity.getUpdatedAt()) <= 0;
```

Comment:
Are you sure about this test ?
Does it mean that if publication is working fine for one of gateway instances, we can consider it as in "sync" state ?